xbmc/platform/darwin/osx/network/NetworkOsx.cpp

```diff
@@ -180,27 +181,45 @@ std::vector<std::string> CNetworkOsx::GetNameServers()
 {
   std::vector<std::string> result;
 
-  FILE* pipe = popen("scutil --dns | grep \"nameserver\" | tail -n2", "r");
-  usleep(100000);
+  FILE* pipe = popen("scutil --dns | grep \"nameserver\" | uniq | tail -n2", "r");
+
   if (pipe)
   {
-    std::vector<std::string> tmpStr;
-    char buffer[256] = {'\0'};
-    if (fread(buffer, sizeof(char), sizeof(buffer), pipe) > 0 && !ferror(pipe))
+    int fileno = ::fileno(pipe);
+    fd_set read_fds;
+    FD_ZERO(&read_fds);
+    FD_SET(fileno, &read_fds);
+
+    struct timeval timeout;
+    timeout.tv_sec = 1; // Set the timeout to 1 second
+    timeout.tv_usec = 0;
+
+    int ready = select(fileno + 1, &read_fds, nullptr, nullptr, &timeout);
+
+    if (ready > 0 && FD_ISSET(fileno, &read_fds))
     {
-      tmpStr = StringUtils::Split(buffer, "\n");
-      for (unsigned int i = 0; i < tmpStr.size(); i++)
+      char buffer[256] = {'\0'};
+      if (fgets(buffer, sizeof(buffer), pipe))
       {
         // result looks like this - > '  nameserver[0] : 192.168.1.1'
-        // 2 blank spaces + 13 in 'nameserver[0]' + blank + ':' + blank == 18 :)
-        if (tmpStr[i].length() >= 18)
-          result.push_back(tmpStr[i].substr(18));
+        // 2 blank spaces + 13 in 'nameserver[0]' + blank + ':' + blank == 18 :
```

Comment:
looks like you destroyed original smiley :)

Reply:
Line is now happy again :)